course_discovery/apps/course_metadata/management/commands/update_course_ai_languages.py

```
@@ -85,8 +85,21 @@ def handle(self, *args, **options):
                 # Remove any keys other than `code` and `label`
                 available_translation_languages = [{'code': lang['code'], 'label': lang['label']} for lang in available_translation_languages]
 
-                # Add the labels for the codes. Currently we set the code as the label. We will be fixing this in a follow-up PR
-                available_transcription_languages = [{'code': lang, 'label': lang} for lang in available_transcription_languages]
+                available_transcription_languages = [
+                    {
+                        "code": lang_code,
+                        "label": (
+                            # Standardizing language codes to match between edx-val and course-discovery:
+                            # - edx-val uses "zh_HANS", "zh_HANT", while course-discovery uses "zh-Hans", "zh-Hant".
+                            # - edx-val uses "en-GB", whereas course-discovery uses "en-gb".
+                            LanguageTag.objects.get(code__iexact=normalized_lang_code).name
+                            if LanguageTag.objects.filter(code__iexact=normalized_lang_code).exists()
+                            else ""
```

Should we raise an exception if we can't find a label?

---

hmm, we can, but do we need to raise an exception here?